CMakeLists.txt

```diff
@@ -1,406 +1,529 @@
-else ()
-    set (DEF_INSTALL_CMAKE_DIR lib/cmake/${VSOMEIP_NAME})
-endif ()
+if(WIN32 AND NOT CYGWIN)
+  set(DEF_INSTALL_CMAKE_DIR CMake)
+else()
+  set(DEF_INSTALL_CMAKE_DIR lib/cmake/${VSOMEIP_NAME})
+endif()
 
-set (INSTALL_CMAKE_DIR ${DEF_INSTALL_CMAKE_DIR} CACHE PATH "Installation directory for CMake files")
+set(INSTALL_CMAKE_DIR
+    ${DEF_INSTALL_CMAKE_DIR}
+    CACHE PATH "Installation directory for CMake files"
+)
 
 # Make relative paths absolute (needed later on)
-foreach (p LIB BIN INCLUDE CMAKE)
-    set (var INSTALL_${p}_DIR)
-    if (NOT IS_ABSOLUTE "${${var}}")
-        set (ABSOLUTE_${var} "${CMAKE_INSTALL_PREFIX}/${${var}}") # Add all targets to the build-tree export set
-    endif ()
-endforeach ()
-
-###################################################################################################
+foreach(p LIB BIN INCLUDE CMAKE)
+  set(var INSTALL_${p}_DIR)
+  if(NOT IS_ABSOLUTE "${${var}}")
+    set(ABSOLUTE_${var} "${CMAKE_INSTALL_PREFIX}/${${var}}") # Add all targets to the build-tree export set
+  endif()
+endforeach()
+
+# #####################################################################
 # Set a default build type if none was specified
+# This is a CMake anti-pattern.
 set(default_build_type "RelWithDebInfo")
 if(NOT CMAKE_BUILD_TYPE)
   message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
-  set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING "Choose the type of build." FORCE)
+  set(CMAKE_BUILD_TYPE
+      "${default_build_type}"
+      CACHE STRING "Choose the type of build." FORCE
+  )
   # Set the possible values of build type for cmake-gui
-  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+  set_property(
+    CACHE CMAKE_BUILD_TYPE
+    PROPERTY STRINGS
+             "Debug"
+             "Release"
+             "MinSizeRel"
+             "RelWithDebInfo"
+  )
 endif()
 
 set(CMAKE_CXX_STANDARD 17)
 
+# Threads
+find_package(Threads REQUIRED)
+
 # OS
-if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-    set(DL_LIBRARY "dl")
+add_library(OS_INTERFACE INTERFACE IMPORTED)
+add_library(RT_INTERFACE INTERFACE IMPORTED)
+add_library(DL_INTERFACE INTERFACE IMPORTED)
+add_library(STACKTRACE_INTERFACE INTERFACE IMPORTED)
 
-if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    # This is only relevant for GCC and causes warnings on Clang
-    set(EXPORTSYMBOLS "-Wl,-export-dynamic -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/exportmap.gcc")
-    set(OS_CXX_FLAGS "${OS_CXX_FLAGS} -pie -Wno-tsan -Wl,-z,relro,-z,now")
+# Boost
+find_package(
+  Boost 1.65 REQUIRED
+  COMPONENTS system thread filesystem
+  OPTIONAL_COMPONENTS stacktrace_basic stacktrace_backtrace stacktrace_addr2line
+)
+
+include(CMakeDependentOption)
+cmake_dependent_option(ENABLE_WERROR "Enable -Werror flag" ON "CMAKE_SYSTEM_NAME MATCHES LINUX" OFF)
+if (ENABLE_WERROR)
+  message(STATUS "Enabling -Werror flag")
+  add_compile_options(-Werror)
 endif()
```

Comment:
why did you separated this flag from the others related to LINUX?

Reply:
The `-Werror` blocks my linux builds :) Honestly I don't think any component should ever include this.  `-Werror` should be up to the toolchain - as the toolchain usually specifies which compile options to use, which warnings to watch for, and whether warnings should be allowed.